### DISCUSSION TITLE
Firefly iii data importer 500 server error

### DISCUSSION THREAD

**Discussion by timothyavid** [firefly-iii/firefly-iii#5558] [Firefly III]
Hello Guys,

I have spent the better part of a day following the documentation. As far as I can get with the Data importer using Salt Edge is next after adding custom configuration .json, then all I get is a 500 Server Error.  :-/

Found this error in my log, have removed keys (HIDDEN)
```
[2022-01-17 18:53:24] production.DEBUG: Final headers for spectre signed POST request: {"App-id":"HIDDEN","Secret":"HIDDEN","Accept":"application/json","Content-type":"application/json","Cache-Control":"no-cache","User-Agent":"FireflyIII Spectre v","Expires-at":1642406184}
[2022-01-17 18:53:25] production.ERROR: Guzzle Exception: Client error: `POST https://www.saltedge.com/api/v5/customers` resulted in a `400 Bad Request` response:

{"error":{"class":"ExpiresAtInvalid","message":"'Expires-At' header is invalid.","documentation_url":"https://docs.salte (truncated...)

{"exception":"[object] (App\\Exceptions\\ImporterHttpException(code: 0): Guzzle Exception: Client error: `POST https://www.saltedge.com/api/v5/customers` resulted in a `400 Bad Request` response:

{\"error\":{\"class\":\"ExpiresAtInvalid\",\"message\":\"'Expires-At' header is invalid.\",\"documentation_url\":\"https://docs.salte (truncated...)

at /var/www/html/app/Services/Spectre/Request/Request.php:275)
[stacktrace]
#0 /var/www/html/app/Services/Spectre/Request/PostCustomerRequest.php(76): App\\Services\\Spectre\\Request\\Request->sendSignedSpectrePost()
#1 /var/www/html/app/Http/Controllers/Import/Spectre/ConnectionController.php(112): App\\Services\\Spectre\\Request\\PostCustomerRequest->post()
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): App\\Http\\Controllers\\Import\\Spectre\\ConnectionController->index()

#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\\Routing\\Controller->callAction()
```
Some Setup info:
```
Now in entrypoint.sh for Firefly III Data Importer
Script: 1.5 (2021-12-05)
User: 'root'
Group: 'root'
Working dir: '/var/www/html'
Build number: 76
Build date: 14-01-2022 20:50:11 CET
Firefly III data importer v0.6.5
PHP: cli 8.0.13 Linux

```
As far as I can tell, I have got everything setup right, but can't get passed this error.

See video: https://i.gyazo.com/2e7de5ac9dc7d745a9b3384f7cde3e52.mp4

both are installed in docker containers under unraid.
What am I doing wrong?

**Comment by JC5** [comment]
Nothing, I think, but I haven't seen it before. I'll check it out, it could be a bug!

**Comment by JC5** [comment]
Could be that the time zone is not configured or configured wrongly. Did you check that already?

**Reply by timothyavid** [reply]
> Could be that the time zone is not configured or configured wrongly. Did you check that already?

Where do I set this? is it part of the docker command?

**Reply by JC5** [reply]
Yes, try `TZ=UTC` or change it to your timezone.

**Reply by timothyavid** [reply]
> 
 No luck still 500 Server Error :/

**Reply by JC5** [reply]
Try it again, and enter this number `"Expires-at":1642406184` in this converter:

https://www.unixtimestamp.com/index.php

The number should be a time about 3 minutes into the future. So if your clock is off by more than 3 minutes, this may lead to errors.

**Reply by timothyavid** [reply]
> 
Just tested that, both server and my computers clock is within 1 minute of that time.

**Reply by timothyavid** [reply]
> Try it again, and enter this number `"Expires-at":1642406184` in this converter:
> 
> https://www.unixtimestamp.com/index.php
> 
> The number should be a time about 3 minutes into the future. So if your clock is off by more than 3 minutes, this may lead to errors.

https://i.gyazo.com/00f08889e0a10a8fedc7ae0d752f5244.mp4

**Reply by JC5** [reply]
OK so less literally this time:

The next time it errors, take the fresh number from the `Expires-at` field at put it in the converter at https://www.unixtimestamp.com/index.php

THEN compare it to your system clock.

**Reply by timothyavid** [reply]
> OK so less literally this time:
> 
> The next time it errors, take the fresh number from the `Expires-at` field at put it in the converter at https://www.unixtimestamp.com/index.php
> 
> THEN compare it to your system clock.

Found the problem, my servers time was not accounting for daylight saving! even though it was set to the right area.
Seems to be working now, just waiting for saltedge to approve my test access

Thanks,

**Reply by JC5** [reply]
Glad to hear it works!
